src/utils/cosmosdb_model_versioned.ts

```diff
@@ -166,6 +173,95 @@ export abstract class CosmosdbModelVersioned<
     );
   }
 
+  public updateTTLForAllVersions(
+    searchKey: DocumentSearchKey<TR, ModelIdKey, PartitionKey>,
+    ttl: NonNegativeNumber
+    // eslint-disable-next-line
+  ): TaskEither<CosmosErrors, TaskEither<never, readonly any[]>> {
+    return pipe(
+      this.findAllVersionsByPartitionKey(searchKey),
```

Comment:
Have you considered the implementation proposed by @BurnedMarshal here?

https://github.com/pagopa/io-functions-commons/pull/294#discussion_r989857265

I guess you don't even need to query for the latest version, just cycle starting from version 0, increment it and stop when the item is not found.

Reply:
In which case do you get version jumps?

Reply:
Manual and scriptef operations (like the data cancellation process) could leave inconsistance data.
We could not enforce this constraints on cosmos, so we have to take care of this use case.

Reply:
This approach doesn't eliminate the possibility of a "version jump" especially on large collection (the time between obtaining a document and patching may occur a few minutes later). The get and update process are not transactional, so we can have a document deletion while the update process is running.

Reply:
IMHO the "version jump" it's a data inconsistency problem that should rise an error on the update process.
I'll think about on the reason of versioned model:
1. Keep the documents readonly
2. Have an history of any change (even for legal purpose)
3. other?
So manual and scripted operations that introduce an inconsistent state should be handled properly, and any cross version operation should fail requiring an administrator intervention to fix it.

Reply:
I agree with @BurnedMarshal

Reply:
Moreover you can use the @BurnedMarshal's algorithm (obtain `max(version)` then delete from there to zero) resunting in the same guarantees (even in case of version jumps) but avoiding to query all revisions.

Reply:
@gunzip for general update operation process data from max to 0 can be used, but for this specific case not, because we are introducing a `ttl` to documents, and a partial update (caused by pagination of max 100 operation in a single transactional batch) can make active an old version.

Reply:
Since it's just an optimization, and it leads to a less readable error handling, can we postpone it when it became necessary? 
For now we will implement @gunzip' suggestion by fetching ids only. 
This could be reasonable?

Reply:
> So manual and scripted operations that introduce an inconsistent state should be handled properly, and any cross version operation should fail requiring an administrator intervention to fix it.
Currenlty, very few peaple have write-access to database: managing this situation in operations could be nasty.